starknet_py/net/account/account.py

```diff
@@ -229,6 +228,16 @@ async def get_balance(
 
         return (high << 128) + low
 
+    async def sign_for_fee_estimate(
+        self, transaction: TypeAccountTransaction
+    ) -> TypeAccountTransaction:
+        version = self.supported_transaction_version + 2**128
+
+        transaction = dataclasses.replace(transaction, version=version)
+        signature = self.signer.sign_transaction(transaction)
+
+        return _add_signature_to_transaction(tx=transaction, signature=signature)
```

Comment:
We use `dataclasses.replace` in line 236, while line 239 uses the same but wrapped into a function.
Let's maybe change `_add_signature_to_transaction` to also contain line 237 (since this logic is also repeated in a few places)

Reply:
Don't want to change the Account logic more than its necessary in this PR, we can create an issue for that if you think its necessary.